index.js

```diff
@@ -43,13 +56,19 @@ LessCompiler.prototype.updateCache = function (srcDir, destDir) {
     less.render(data, lessOptions)
       .then(function (output) {
 
-        fs.writeFile(destFile, output.css, { encoding: 'utf8' }, function (err) {
-          if (err) {
-            return reject(err);
-          }
+	  function fileCallback (err) {
+	    if (err) {
+	      return reject(err);
+	    }
+
+	    return resolve(output);
+	  }
+
+    fs.writeFile(destFile, output.css, { encoding: 'utf8' }, fileCallback);
 
-          return resolve(output);
-        });
+	  if (lessOptions.hasOwnProperty('sourceMap')) {
+	    writeSourcemapFile(destFile, output.map, lessOptions.sourceMap, fileCallback);
```

Comment:
We are resolving the same promise twice.  Might be better to use push these async methods to some promise array and use RSVP.all to resolve the containing Promise.